---
title: "Gonzalez_Schrode2022_Analysis"
output: html_document
---

#Load libraries 
```{r echo = FALSE}
library(Seurat)
library(sctransform)
library(SeuratWrappers)
library(dplyr)
library(ggplot2)
library(patchwork)
library(tidyverse)
library(future)
plan("multiprocess", workers = 5)
plan()
library(VennDiagram)
library(EnhancedVolcano)

```

*** 

# Normal Development samples without RA injected embryos 

Read in objects downloaded for individual time points 
```{r}
cc <- readRDS("~/Project/Seurat Objects/cc.rds")
pht <- readRDS("~/Project/Seurat Objects/pht.rds")
ht <- readRDS("~/Dropbox/David/Single cell analysis/A_V v2/HT/Seurat Objects/ht.rds")
```


```{r}
DimPlot(object = cc, reduction = 'umap', label = TRUE)
DimPlot(object = pht, reduction = 'umap', label = TRUE)
DimPlot(object = ht, reduction = 'umap', label = TRUE)

```


Next we wanted to find populations of cardiac cells, as well as endocardial cell types and pharyngeal progenitors within each of the different stage s

## CC Stage Marker Exploration 

```{r fig.height = 10, fig.width=15}
#Mesodermal Markers, Endoderm Markers, Ectoderm Markers
FeaturePlot(object = cc, features = c("Nkx2-5", "Isl1", "Hcn4", "Epcam", "Pdgfra", "Tbx1", "Foxf1", "Aldh1a2", "Sox2", "Tnnt2", "EGFP", "Crabp1"), min.cutoff = "q10", order = TRUE )

#Endocardial Markers, Lateral Plate Mesoderm and Paraxial Mesoderm Markers
FeaturePlot(object = cc, features = c("Pecam1", "Egfl7", "Tbx18", "Wt1", "Sox17", "Krt8", "Pmp22", "Fst", "Foxa2", "Foxc2", "Pax9"), min.cutoff = "q10", order = TRUE)

FeaturePlot(object = cc, features = c("Irx4", "Kcna5", "Tbx18", "Wt1", "Sox17", "Krt8", "Pmp22", "Fst", "Foxa2", "Foxc2", "Pax9"), min.cutoff = "q10", order = TRUE)


```


A few clusters weren't labeled by the markers of interest above but clustered closely to other markers, so we performed and did some differential expression analysis and plugged those into GO/KEGG pathway analysis packages such as enrichr to see what they would give. Summary of those findings is found below


```{r}
#example 
cc.12.markers <- FindMarkers(object = cc, ident.1 = 12, min.pct = 0.25, only.pos = TRUE)

```

Clusters of interest by stage for documentation:
CC.10 = hemoglobin expressing cells, 
CC.1 = Sox2 high, expressing 
CC.12 = High markers having to do with cell-adhesion and sarcomere, lipid raft assembly etc, so was included 
CC.11 = Sox2 high as well with not enough mesodermal expression and thus was removed as likely ectoderm derived from NC lineage 


*** FINAL SUBSET CHOSEN FOR CARDIAC CRESCENT: 7,9,12,13,15,17


## Primitive Heart Tube Marker Exploration 

```{r fig.height =5, fig.width=8}
#Mesodermal Markers, Endoderm Markers, Ectoderm Markers
FeaturePlot(object = pht, features = c("Nkx2-5", "Isl1", "Sox2","Epcam", "Pdgfra", "Foxc1", "Foxf1", "Aldh1a2", "EGFP", "Tnnt2", "Sox9", "Crabp1"), min.cutoff = "q10", order = TRUE)

#Endocardial Markers, Lateral Plate Mesoderm and Paraxial Mesoderm Markers, others
FeaturePlot(object = pht, features = c("EGFP", "Egfl7", "Tbx18", "Wt1", "Upk3b", "Krt8", "Pmp22", "Fst", "Tcf15", "Foxc2", "Hba-x"), order = TRUE)


```


```{r}
pht.8.markers <- FindMarkers(object = pht, ident.1 = 8, min.pct = 0.25, only.pos = TRUE)

```


Clusters of interest by stage for documentation:
PHT.5 = High levels of mitochondrial genes and few differentially expressed genes, likely low quality input cells 
PHT.0 = Only two differentialy expressed genes, Twist1 and a ribosomal RNA so this should not be included 


*** Final Clusters Selected for PHT: 6,7,8,9,12,17,18 *** 

## HT Annotation 

Previous Clusters to subset/choose for subclustering and integration based on the above: 4,8,17,2,15,18, 21, 15

```{r fig.height = 10, fig.width=15}
#Mesodermal Markers, Endoderm Markers, Ectoderm Markers
FeaturePlot(object = ht2, features = c("Nkx2-5", "Isl1", "Sox2","Epcam", "Pdgfra", "Sox17", "Foxa2", "Aldh1a2", "EGFP", "Tnnt2", "Irx4", "Kcna5"), min.cutoff = "q10", order = TRUE)

#Endocardial Markers, Lateral Plate Mesoderm and Paraxial Mesoderm Markers, others
FeaturePlot(object = ht2, features = c("Pecam1", "Egfl7", "Tbx18", "Wt1", "Upk3b", "Krt8", "Pmp22", "Fst", "Tcf15", "Foxc2", "Hba-x"), min.cutoff = "q10", order = TRUE)

```

Example code for further exploration of other clusters 

```{r}
ht.12.markers <- FindMarkers(object = ht2, ident.1 = 12, min.pct = 0.25, only.pos = TRUE)

```

Clusters of interest by stage for documentation:
HT.21 = High levels of hemoglobin expressing genes (#7 is also hemoglobin expressing)
HT.1 = PPAR-gamma signaling, cholesterol biosynthesis, was included for the sake of completeness 
HT.5 = Mostly rna binding proteins, likely not high quality cells and thus excluded
HT.12= Splicing and ribosomal processes, was excluded for this reason 


** Final subset for HT: 2,4,8,9,14,15,17,18


##Subsetting cardiac (and associated lineages) cell types into new objects 


```{r}
cc_card <- subset(cc, idents = c(7,9,12,13,15,17))
pht_card <- subset(pht, idents = c(6,7,8,9,12,17,18))
ht_card <- subset(ht, idents = c(2,4,8,9,14,15,17,18))
```


***

We now wanted to reanalyze the objects for just the cardiac regions of interest for the individual samples 

## Normalizing data through scTransform 

```{r scTransform}
cc_card <- SCTransform(object = cc_card, verbose = FALSE)
pht_card <- SCTransform(object = pht_card, verbose = FALSE)
ht_card <- SCTransform(object = ht_card, verbose = FALSE)
```



## Perform linear dimensional reduction


```{r pca,results='hide'}
cc_card <- RunPCA(object = cc_card, features = VariableFeatures(object = cc_card))
pht_card <- RunPCA(object = pht_card, features = VariableFeatures(object = pht_card))
ht_card <- RunPCA(object = ht_card, features = VariableFeatures(object = ht_card))

```


```{r pca_viz, message=TRUE}
# Examine and visualize PCA results a few different ways
print(x = cc_card[['pca']], dims = 1:5, nfeatures = 5)
VizDimLoadings(object = cc_card, dims = 1:2, reduction = 'pca')
DimPlot(object = cc_card, reduction = 'pca')
```


```{r pca_viz, message=TRUE}
# Examine and visualize PCA results a few different ways
print(x = pht_card[['pca']], dims = 1:5, nfeatures = 5)
VizDimLoadings(object = pht_card, dims = 1:2, reduction = 'pca')
DimPlot(object = pht_card, reduction = 'pca')
```

```{r pca_viz, message=TRUE}
# Examine and visualize PCA results a few different ways
print(x = ht_card[['pca']], dims = 1:5, nfeatures = 5)
VizDimLoadings(object = ht_card, dims = 1:2, reduction = 'pca')
DimPlot(object = ht_card, reduction = 'pca')
```


## Determine the 'dimensionality' of the dataset 


```{r elbow_plot}
ElbowPlot(object = cc_card)
ElbowPlot(object = pht_card)
ElbowPlot(object = ht_card)

```

## Cluster the cells

```{r cluster, fig.height=5, fig.width=7}
cc_card <- FindNeighbors(object = cc_card, dims = 1:30)
cc_card <- FindClusters(object = cc_card, resolution = 1.5)

pht_card <- FindNeighbors(object = pht_card, dims = 1:30)
pht_card <- FindClusters(object = pht_card, resolution = 0.5)

ht_card <- FindNeighbors(object = ht_card, dims = 1:30)
ht_card <- FindClusters(object = ht_card, resolution = 1.2)

```


## Run non-linear dimensional reduction (UMAP/tSNE)


```{r tsne, fig.height=5, fig.width=7}

cc_card <- RunUMAP(object = cc_card, dims = 1:30)
pht_card <- RunUMAP(object = pht_card, dims = 1:30)
ht_card <- RunUMAP(object = ht_card, dims = 1:30)

```



*** 

# Cardiac Crescent Sample Alone 

Note: for the following section (until the next header break) the same analysis flow can be repeated, using the PHT and HT samples in place of the heart tube and primitive heart tube samples. These have been ommitted for brevity, but can be analyzed further by the user in the same fashion by changing the variables as needed. 

## Cell Cycle Scoring 

```{r}
exp.mat <- read.table(file = "~/Downloads/cell_cycle_vignette_files/nestorawa_forcellcycle_expressionMatrix.txt", header = TRUE, as.is = TRUE, row.names = 1)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

cc_card <- CellCycleScoring(cc_card, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

head(cc_card@meta.data)
```

```{r}
DimPlot(cc_card, group.by = "Phase")
```

```{r message=FALSE, warning=FALSE, include=FALSE}
# determine total cell number
Idents(cc_card)= "Phase"

G1=length(WhichCells(cc_card, idents="G1"))
G2M=length(WhichCells(cc_card, idents="G2M"))
S=length(WhichCells(cc_card, idents="S"))

```

```{r message=FALSE, warning=FALSE, include=FALSE}
# normalize frequencies by total cell number
freq=table(cc_card@meta.data$Phase, cc_card@meta.data$seurat_clusters)
freq1=freq
freq1[1,]=freq[1,]/G1
freq1[2,]=freq[2,]/G2M
freq1[3,]=freq[3,]/S
# calculate proportions
p=prop.table(freq1, 2)
p=as.data.frame(p)
```


```{r message=FALSE, warning=FALSE, include=FALSE}
q=qplot(x = Var2, y = Freq, fill = Var1, data = p, geom = "col") +
  #scale_fill_manual(values = c("red", "green", "blue")) +
  labs(x="Cell Phase", fill="Phase") +
  coord_flip()+
  theme_bw(14)


plot(q)
```

## Scaling RNA assay for cc_card in order to perform DE appropriately 

```{r}
DefaultAssay(cc_card) = "RNA"
Idents(cc_card) = "seurat_clusters"

cc_card <- NormalizeData(cc_card, verbose = FALSE)
cc_card <- FindVariableFeatures(cc_card, selection.method = "vst", nfeatures = 3000)
cc_card <- ScaleData(cc_card, verbose = FALSE)
```



## Cardiac Crescent Marker Visualization 

```{r fig.height = 10, fig.width=15}
#Mesodermal Markers, Endoderm Markers, Ectoderm Markers
FeaturePlot(object = cc_card, features = c("Pmp22", "Nkx2-5", "Tnnt2", "Hcn4", "Tbx5", "Isl1", "Tbx1", "Aldh1a2", "Foxf1", "EGFP"), min.cutoff = "q10", order = TRUE) & scale_color_gradientn(colors = c("grey80", viridis::mako(n = 50, direction = -1)))

```


## Manual Annotation of CLusters 



## Stashing separate "grouped_cell_type" metadata classifier in order to group separate clusters of shared identity together 

```{r message=FALSE, warning=FALSE, include=FALSE}
levels(cc_card$seurat_clusters)

Idents(cc_card) = "seurat_clusters"
cc_card <- RenameIdents(cc_card, 
                              `0` = "MP", 
                              `1` = "pSHF", 
                              `2` = "LPM", 
                              `3` = "SM", 
                              `4` = "EC", 
                              `5` = "aSHF", 
                              `6` = "EC", 
                              `7` = "FHF", 
                              `8` = "aSHF", 
                              `9` = "LPM", 
                              `10` = "LPM", 
                              `11` = "LPM", 
                              `12` = "pSHF",
                              `13` = "FHF",
                              `14` = "LPM",
                              `15` = "MP",
                              `16` = "aSHF",
                              `17` = "SM",
                              `18` = "pSHF")

cc_card[["grouped_cell_type"]] <- Idents(object = cc_card)
```

## Annotation of cell type but still preserving separate cluster identity 

```{r message=FALSE, warning=FALSE, include=FALSE}
levels(cc_card$seurat_clusters)

Idents(cc_card) = "seurat_clusters"
cc_card <- RenameIdents(cc_card, 
                              `0` = "0 mesodermal progenitor", 
                              `1` = "1 pSHF", 
                              `2` = "2 LPM", 
                              `3` = "3 somitic mesoderm", 
                              `4` = "4 endothelial/endocardial cells", 
                              `5` = "5 aSHF", 
                              `6` = "6 hemangioblasts", 
                              `7` = "7 FHF", 
                              `8` = "8 aSHF", 
                              `9` = "9 mesodermal progenitor", 
                              `10` = "10 LPM", 
                              `11` = "11 LPM", 
                              `12` = "12 pSHF",
                              `13` = "13 FHF",
                              `14` = "14 LPM",
                              `15` = "15 myocardial cells",
                              `16` = "16 aSHF derivatives",
                              `17` = "17 somitic mesoderm",
                              `18` = "18 pSHF")

cc_card[["cell.type.anno.man"]] <- Idents(object = cc_card)
```



```{r}
saveRDS(cc_card, file = "~/Dropbox/David/Single cell analysis/A_V v2/BioRxiv Manuscript/Seurat Objects/cc_card.rds")

```



## Differential expression analysis of EGFP+/- cells at CC stage 

```{r}
# create meta data column for cell ID and new EGFP cell category
cc_card@meta.data$cell=rownames(cc_card@meta.data)

cc_card@meta.data$EGFPexpression="neg"


## determine background cutoff
# visually inspect expression range
FeaturePlot(cc_card, features=c("EGFP"), order = T, min.cutoff = "q10", label=T, coord.fixed = T, combine=T)

# test cell number for different lower expression limits
length(WhichCells(object = cc_card, expression = EGFP > 1))
# save cell IDs that express EGFP
tmp=WhichCells(object = cc_card, expression = EGFP > 1) 


# set those cells to EGFPexpression positive
cc_card@meta.data[cc_card@meta.data$cell %in% tmp, ]$EGFPexpression <- "pos"


# confirm proper annotation in UMAP plot
DimPlot(cc_card, reduction = "umap", group.by="EGFPexpression", cols = c("grey90", "green"))

DimPlot(cc_card, reduction = "umap", label = TRUE)


# set default identity to new EGFP expression category
Idents(cc_card) = "EGFPexpression"
DefaultAssay(cc_card) = "RNA"


# run differential expression 
# set only.pos = T, because there are only two groups in this category
# Wilcoxon rank-sum test is the default test Seurat uses and quite generic, you can't really go wrong with it. People often prefer a negative binomial linear model though. It fits the distribution of UMI based scRNA-seq pretty well and often delivers more DEGs.

cc_card.EGFP.DE = FindMarkers(cc_card, ident.1 = "pos", ident.2 = "neg", only.pos = FALSE, test.use = "negbinom")

write.csv(cc_card.EGFP.DE, file = "~/Dropbox (DuboisLab)/David/Single cell analysis/A_V v2/BioRxiv Manuscript/Differential Expression Tests/CC_card_EGFPpos_vs_EGFPneg.csv")

cc_card.EGFP.DE <- cc_card.EGFP.DE[which(cc_card.EGFP.DE$p_val_adj<=0.05),]


```


```{r fig.height=10, fig.width=10}
library(EnhancedVolcano)
EnhancedVolcano(cc_card.EGFP.DE,
                lab = rownames(cc_card.EGFP.DE),
               #selectLab = c("Wnt2", "Gsk3b", "Tnnt2", "Sfrp1", "Wnt5a"),
                x = 'avg_log2FC', 
                y = 'p_val', 
                xlim = c(-4,4), 
                #title = 'Ventricular (Cluster 17&2) vs Atrial (Cluster 4)', 
                pCutoff = 1e-2, 
                FCcutoff = 0.2, 
                pointSize = 2.0, 
                labSize = 7.0, 
                gridlines.major = FALSE, 
                gridlines.minor = FALSE, 
                legendLabels = FALSE)
```



*** 

# Integration of Individual Time Point 

Next we wished to integrate the cardiac subset data from all three time points (CC, PHT, HT) into one Seurat object. This was performed using Seurat's standard integration workflow as detailed below. 



```{r}
sample.list <- c(cc_card, pht_card, ht_card)

options(future.globals.maxSize = 73400032000)

features <- SelectIntegrationFeatures(object.list = sample.list, nfeatures = 5000)

sample.list <- PrepSCTIntegration(object.list = sample.list, anchor.features = features, verbose = FALSE)

anchors <- FindIntegrationAnchors(object.list = sample.list, normalization.method = "SCT", anchor.features = features, verbose = F)

TP_merge_cardiac <- IntegrateData(anchorset = anchors, normalization.method = "SCT", verbose = F)

#Run Dimensionality Reduction etc 
TP_merge_cardiac <- RunPCA(TP_merge_cardiac, verbose = FALSE)

TP_merge_cardiac <- RunUMAP(TP_merge_cardiac, reduction = "pca", dims = 1:30)

TP_merge_cardiac <- FindNeighbors(object = TP_merge_cardiac, dims = 1:30)
TP_merge_cardiac <- FindClusters(object = TP_merge_cardiac, resolution = 1.2)

```



```{r fig.height=5, fig.width=7}
Idents(TP_merge_cardiac) = "orig.ident"

DimPlot(object = TP_merge_cardiac, reduction = 'umap', label = TRUE)

DimPlot(object = TP_merge_cardiac, reduction = 'umap', label = FALSE, group.by = "orig.ident", cols = c("grey", "orange", "blue"))
DimPlot(object = TP_merge_cardiac, reduction = 'umap', label = FALSE, split.by = "orig.ident", cols = c("grey", "orange", "blue"))

```



## Quantification of Time Point Contribution 

```{r message=FALSE, warning=FALSE, include=FALSE}
# determine total cell number
Idents(TP_merge_cardiac)= "orig.ident"


CC=length(WhichCells(TP_merge_cardiac, idents="CC"))
PHT=length(WhichCells(TP_merge_cardiac, idents="PHT"))
HT=length(WhichCells(TP_merge_cardiac, idents="HT"))

```

```{r message=FALSE, warning=FALSE, include=FALSE}
# normalize frequencies by total cell number
freq=table(TP_merge_cardiac@meta.data$orig.ident, TP_merge_cardiac@meta.data$seurat_clusters)
freq1=freq
freq1[1,]=freq[1,]/CC
freq1[2,]=freq[2,]/PHT
freq1[3,]=freq[3,]/HT
# calculate proportions
p=prop.table(freq1, 2)
p=as.data.frame(p)
```


```{r message=FALSE, warning=FALSE, include=FALSE}
q=qplot(x = Var2, y = Freq, fill = Var1, data = p, geom = "col") +
  scale_fill_manual(values = c("grey", "orange", "blue")) +
  labs(x="Time Point Contribution", fill="orig.ident") +
  coord_flip()+
  theme_bw(14)


plot(q)
```



## Cell Cylce Scoring 

```{r}
exp.mat <- read.table(file = "~/Downloads/cell_cycle_vignette_files/nestorawa_forcellcycle_expressionMatrix.txt", header = TRUE, as.is = TRUE, row.names = 1)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

TP_merge_cardiac <- CellCycleScoring(TP_merge_cardiac, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

RidgePlot(TP_merge_cardiac, features = c("Tnnt2", "Pmp22", "EGFP", "Myl2"), ncol = 2)

head(TP_merge_cardiac@meta.data)
```

```{r}
DimPlot(TP_merge_cardiac, group.by = "Phase")
```

```{r message=FALSE, warning=FALSE, include=FALSE}
# determine total cell number
Idents(TP_merge_cardiac)= "Phase"

G1=length(WhichCells(TP_merge_cardiac, idents="G1"))
G2M=length(WhichCells(TP_merge_cardiac, idents="G2M"))
S=length(WhichCells(TP_merge_cardiac, idents="S"))

```

```{r message=FALSE, warning=FALSE, include=FALSE}
# normalize frequencies by total cell number
freq=table(TP_merge_cardiac@meta.data$Phase, TP_merge_cardiac@meta.data$seurat_clusters)
freq1=freq
freq1[1,]=freq[1,]/G1
freq1[2,]=freq[2,]/G2M
freq1[3,]=freq[3,]/S
# calculate proportions
p=prop.table(freq1, 2)
p=as.data.frame(p)
```


```{r message=FALSE, warning=FALSE, include=FALSE}
q=qplot(x = Var2, y = Freq, fill = Var1, data = p, geom = "col") +
  #scale_fill_manual(values = c("red", "green", "blue")) +
  labs(x="Cell Phase", fill="Phase") +
  coord_flip()+
  theme_bw(14)


plot(q)
```



## Normalization and scaling of markers for visualization and differential expression  

```{r}
DefaultAssay(TP_merge_cardiac) = "RNA"
Idents(TP_merge_cardiac) = "seurat_clusters"

TP_merge_cardiac <- NormalizeData(TP_merge_cardiac, verbose = FALSE)
TP_merge_cardiac <- FindVariableFeatures(TP_merge_cardiac, selection.method = "vst", nfeatures = 3000)
TP_merge_cardiac <- ScaleData(TP_merge_cardiac, verbose = FALSE)

```



## Visualizaion of Markers 
```{r fig.height=10, fig.width=15}
DefaultAssay(TP_merge_cardiac) = "RNA"

FeaturePlot(object = TP_merge_cardiac, features = c("Nkx2-5", "Isl1", "Hcn4", "Pdgfra", "Hand1", "Foxf1", "Aldh1a2", "EGFP", "Tnnt2", "Epcam", "Tbx1", "Tbx18"), min.cutoff = "q10")

FeaturePlot(object = TP_merge_cardiac, features = c("Tnnt2", "Myl2", "Myl7", "Kcna5", "Nppa", "Irx4", "Nr2f2", "EGFP", "Shox2", "Wt1", "Tbx5", "Rspo3"), min.cutoff = "q10")

FeaturePlot(object = TP_merge_cardiac, features = c("Pax3", "Tcf15", "Sox9", "Fst", "Meox1", "Foxc2", "Six2", "Cyp26a1", "Tbx1", "Tbx18", "Tbx5", "Rspo3"), min.cutoff = "q10")

FeaturePlot(object = TP_merge_cardiac, features = c("Pecam1", "Egfl7", "Nfatc1", "Etv2", "Wt1", "Sox2", "Krt8", "Dlx5", "Fst", "Dlx2", "Kdr", "Hand1"), min.cutoff = "q10")

```



```{r fig.height=8, fig.width=20}
VlnPlot(object = TP_merge_cardiac, features = c("Rara", "Rarb", "Rarg", "Hand1", "Cck", "Pln", "Smoc2", "Acta2", "Stard10", "Lbh", "Myh7", "Bmp4", "Fgf8"), pt.size = 0)

VlnPlot(object = TP_merge_cardiac, features = c("Pecam1", "Egfl7", "Nfatc1", "Tbx18", "Wt1", "Sox2", "Krt8", "Pmp22", "Fst", "Rgs5", "Kdr", "Hand1"), pt.size = 0)
```





## Differential Expression analysis for Identification of Clusters 

The same general analysis pipeline below was used for mulitple comparisons in this work, including pairwise comparisons of atrial vs ventricular cells, as well as analysis of individual clusters of interest during the annotation process etc. In order to better understand the results from these analyses, differentially expressed gene lists were also fed through GO/KEGG and gene set enrichment analsysis packages detailed elsewhere in order to more completely understand the processes and signaling pathways defining each cluster. 

```{r fig.height=5, fig.width=5}

Idents(TP_merge_cardiac)= "seurat_clusters"
DefaultAssay(TP_merge_cardiac) <- "RNA"


#Find all markers for each cluster
TP_merge_cardiac.markers <- FindAllMarkers(object = TP_merge_cardiac, min.pct = 0.25, only.pos = FALSE)



## example code for comparison used of atrial vs ventricular cells. The code can be modified for other comparisons of interest as needed folliwng the Seurat DE vignette
TP_merge_cardiac.av.markers <- FindMarkers(object = TP_merge_cardiac, ident.1 = c(2,17), ident.2 = 4, min.pct = 0.25, only.pos = FALSE, test.use = "negbinom")
```


```{r fig.height=10, fig.width=10}
library(EnhancedVolcano)
EnhancedVolcano(TP_merge_cardiac.av.markers,
                lab = rownames(TP_merge_cardiac.av.markers),
               #selectLab = c("Wnt2", "Gsk3b", "Tnnt2", "Sfrp1", "Wnt5a"),
                x = 'avg_log2FC', 
                y = 'p_val', 
                xlim = c(-5,5), 
                title = 'Ventricular (Cluster 17&2) vs Atrial (Cluster 4)', 
                pCutoff = 1e-3, 
                FCcutoff = 0.2, 
                pointSize = 2.0, 
                labSize = 7.0, 
                gridlines.major = FALSE, 
                gridlines.minor = FALSE, 
                legendLabels = FALSE)
```



## Subset of cluster 5 cells 

As a further example of how individual populations were subclustered and futher explored, the code below demonstrates how we subclustered a cluster of interest (cluster 5 belonging to an aSHF derivative) to uncover additional heterogeneity present in this population. This was done several times as we were annotating populations, and selected parts of that analysis are included in the figures with the publication of this work. 

```{r}
Idents(TP_merge_cardiac) = "seurat_clusters"
cluster5 <- subset(TP_merge_cardiac, idents = 5)
```


```{r}
cluster5 <- RunUMAP(cluster5, reduction = "pca", dims = 1:30)

```

```{r}
DefaultAssay(cluster5) = "integrated"
cluster5 <- FindNeighbors(object = cluster5, dims = 1:30)
cluster5 <- FindClusters(object = cluster5, resolution = 0.5)

p1 <- DimPlot(cluster5, label = TRUE)

```

```{r fig.height=10, fig.width=15}
Idents(cluster5) = "seurat_clusters"
VlnPlot(cluster5, features = c("Tnnt2","Rgs5","Myl2", "Tdgf1", "3632451O06Rik", "Crabp1", "Fhl1", "Rspo3", "Flna"), pt.size = 0)
```


```{r}
DefaultAssay(cluster5) = "RNA"
cluster5.markers <- FindMarkers(cluster5, ident.1 = 1, min.pct = 0.25, only.pos = FALSE)
```


## Assigning of cluster identity to cell types 

After subclustering individual clusters to better understand their identity and perfroming differential expression analysis and marker exploration for the individiual populations. For the purposes of classification and data display, we created several different categories of classification data with varying degrees of specificity. 


```{r message=FALSE, warning=FALSE, include=FALSE}
levels(TP_merge_cardiac$seurat_clusters)

Idents(TP_merge_cardiac) = "seurat_clusters"
TP_merge_cardiac <- RenameIdents(TP_merge_cardiac, 
                              `0` = "0 myocardial precursor", 
                              `1` = "1 posterior second heart field", 
                              `2` = "2 RV", 
                              `3` = "3 hematoendothelial progenitor", 
                              `4` = "4 atrial CM", 
                              `5` = "5 aSHF myocardial cells/OFT", 
                              `6` = "6 lateral plate mesoderm", 
                              `7` = "7 atrioventricular canal", 
                              `8` = "8 differentiating aSHF", 
                              `9` = "9 endothelial cells", 
                              `10` = "10 endothelial cells", 
                              `11` = "11 paraxial mesoderm", 
                              `12` = "12 sinus venosus",
                              `13` = "13 endothelial cells",
                              `14` = "14 differentiating pSHF",
                              `15` = "15 cardiac committed mesoderm",
                              `16` = "16 endothelial cells",
                              `17` = "17 LV",
                              `18` = "18 neural crest",
                              `19` = "19 anterior second heart field",
                              `20` = "20 paraxial mesoderm",
                              `21` = "21 posterior second heart field",
                              `22` = "22 anterior second heart field",
                              `23` = "23 endocardial cells",
                              `24` = "24 hematoendothelial progenitor",
                              `25` = "25 epicardium",
                              `26` = "26 lateral plate mesoderm")

TP_merge_cardiac[["cell.type.anno.man"]] <- Idents(object = TP_merge_cardiac)
Idents(TP_merge_cardiac) = "seurat_clusters"
```



```{r message=FALSE, warning=FALSE, include=FALSE}
levels(TP_merge_cardiac$seurat_clusters)

Idents(TP_merge_cardiac) = "seurat_clusters"
TP_merge_cardiac <- RenameIdents(TP_merge_cardiac, 
                              `0` = "Myocard Prog",
                              `1` = "pSHF", 
                              `2` = "RV", 
                              `3` = "HP", 
                              `4` = "Atria", 
                              `5` = "aSHF/OFT", 
                              `6` = "LPM", 
                              `7` = "AVC", 
                              `8` = "aSHF", 
                              `9` = "EC", 
                              `10` = "EC", 
                              `11` = "SM", 
                              `12` = "SV",
                              `13` = "EC",
                              `14` = "pSHF",
                              `15` = "LPM",
                              `16` = "EC",
                              `17` = "LV",
                              `18` = "NC",
                              `19` = "aSHF",
                              `20` = "SM",
                              `21` = "pSHF",
                              `22` = "aSHF",
                              `23` = "EC",
                              `24` = "EC",
                              `25` = "Epicardium",
                              `26` = "LPM")

TP_merge_cardiac[["grouped.cell.type.expand"]] <- Idents(object = TP_merge_cardiac)
Idents(TP_merge_cardiac) = "seurat_clusters"


```


```{r message=FALSE, warning=FALSE, include=FALSE}
levels(TP_merge_cardiac$seurat_clusters)

Idents(TP_merge_cardiac) = "seurat_clusters"
TP_merge_cardiac <- RenameIdents(TP_merge_cardiac, 
                              `0` = "Myocardial Precursor",
                              `1` = "SHF", 
                              `2` = "Ventricular Cardiomyocyte" 
                              `3` = "Hematoendothelial progenitor", 
                              `4` = "Atrial Cardiomyocyte" 
                              `5` = "Differentiating cardiac progenitor", 
                              `6` = "Lateral Plate Mesoderm", 
                              `7` = "Atrioventricular canal", 
                              `8` = "SHF", 
                              `9` = "Endothelial cell", 
                              `10` = "Endothelial cell", 
                              `11` = "Somitic Mesoderm", 
                              `12` = "Differentiating cardiac progenitor",
                              `13` = "Endothelial cell",
                              `14` = "Differentiating cardiac progenitor",
                              `15` = "Lateral Plate Mesoderm",
                              `16` = "Endothelial cell",
                              `17` = "Ventricular Cardiomyocyte",
                              `18` = "Neural Crest",
                              `19` = "Pharyngeal Mesoderm",
                              `20` = "Somitic Mesoderm",
                              `21` = "SHF",
                              `22` = "SHF",
                              `23` = "Endocardial cell",
                              `24` = "Hematoendothelial progenitor",
                              `25` = "Epicardium",
                              `26` = "Lateral Plate Mesoderm")

TP_merge_cardiac[["grouped.cell.type.v2"]] <- Idents(object = TP_merge_cardiac)
Idents(TP_merge_cardiac) = "seurat_clusters"


```


